contracts/deposit/src/lib.rs

```diff
@@ -0,0 +1,156 @@
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
+
+use ink_core::{
+    memory::format,
+    memory::string::String,
+    storage::{self,Vec,Flush}
+};
+use ink_lang::contract;
+
+use parity_codec::{Encode,Decode};
+
+type RangeNumber = u128;
+type BlockNumber = u128;
+type ChallengeNumber = u128;
+
+#[derive(Clone,Encode,Decode,Default,PartialEq,Eq)]
+#[cfg_attr(feature="std",derive(Debug))]
+pub struct Range{
+    start : RangeNumber,
+    end : RangeNumber,
+}
+
+#[derive(Encode,Decode)]
+#[cfg_attr(feature="std",derive(Debug))]
+pub struct StateObject{
+    id : String,
+    predicate : AccountId,
+    data: Vec<u8>,
+}
+
+#[derive(Encode,Decode)]
+#[cfg_attr(feature="std",derive(Debug))]
+pub struct StateUpdate {
+    range : Range,
+    state_object : StateObject,
+    plasma_contract : AccountId,
+    plasma_block_number : BlockNumber,
+}
+
+#[derive(Encode,Decode)]
+#[cfg_attr(feature="std",derive(Debug))]
+pub struct Checkpoint {
+    state_update : StateUpdate,
+    sub_range : Range,
+}
+
+#[derive(Encode,Decode)]
+#[cfg_attr(feature="std",derive(Debug))]
+pub struct CheckpointStatus {
+    challengeable_until : BlockNumber,
+    outstanding_challenges : ChallengeNumber,
+}
+
+#[derive(Encode,Decode)]
+#[cfg_attr(feature="std",derive(Debug))]
+pub struct Challenge {
+    challenged_checkpoint : Checkpoint,
+    challenging_checkpoint : Checkpoint,
+}
+
+contract! {
+    #![env = ink_core::env::DefaultSrmlTypes]
+
+    event CheckpointStarted{
+        checkpoint : Checkpoint,
+        challengeable_until: BlockNumber,
+    }
+
+    event CheckpointChallenged{
+        challenge : Challenge,
+    }
+
+    event CheckpointFinalized{
+        checkpoint: Hash,
+    }
+
+    event ExitStarted{
+        exit : Hash,
+        redeemable_after : BlockNumber,
+    }
+
+    event ExitFinalized{
+        exit : CheckPoint,
+    }
+
+    struct Deposit {
+        //constant values
+
+        //MUST be an adress of ERC20 token
+        TOKEN_ADDRES : storage::Value<AccountId>,
+        CHALLENGE_PERIOD : storage::Value<BlockNumber>,
+        EXIT_PERIOD : storage::Value<BlockNumber>,
+
+        //changable values
+        total_deposited : storage::Value<Range>,
+        checkpoints : storage::HashMap<Hash,Checkpoint>,
+        deposited_ranges : storage::HashMap<RangeNumber, Range>,
+        exit_redeemable_after : storage::HashMap<Hash,BlockNumber>,
+        challenges : storage::HashMap<Hash,bool>,
+
+        //delete later
+        value: storage::Value<bool>,
```

Comment:
now delete and comment out `//self.value` function.

Reply:
deleted!